patroni/validator.py

```diff
@@ -938,11 +938,13 @@ def validate_watchdog_mode(value: Any) -> None:
     "name": str,
     "scope": str,
     Optional("log"): {
+        Optional("type"): EnumValidator(('plain', 'json'), case_sensitive=True, raise_assert=True),
         Optional("level"): EnumValidator(('DEBUG', 'INFO', 'WARN', 'WARNING', 'ERROR', 'FATAL', 'CRITICAL'),
                                          case_sensitive=True, raise_assert=True),
         Optional("traceback_level"): EnumValidator(('DEBUG', 'ERROR'), raise_assert=True),
-        Optional("format"): str,
+        Optional("format"): Or(str, [Or(str, dict)]),
```

Comment:
> Please also update the validator.py:
> 
> https://github.com/zalando/patroni/blob/5c3e1a693e2217e990ae42cd27c1d9795df38d68/patroni/validator.py#L940-L951

It should work but in my tests it cannot validate format when it is list. I also looked at `initdb` schema and found that it is same as this.

Reply:
I think the validator isn't smart enough to handle nested `Or`.
You can create a custom function that will validate the value.